Panic in WmiMethod.Execute when a WMI method does not have any input parameter

These codes seem to panic when calling a method UpdateHostStorageCache on MSFT_StorageSetting which have no input parameters.

```
		inparamsRaw, err := rawMethod.ToIDispatch().GetProperty("InParameters")
		if err != nil {
			return nil, err
		}
		defer inparamsRaw.Clear()

		inparams, err := oleutil.CallMethod(inparamsRaw.ToIDispatch(), "SpawnInstance_")
		if err != nil {
			return nil, err
		}
		defer inparams.Clear()

		for _, inp := range inParam {
			// 	log.Printf("InParam [%s]=>[%+v]\n", inp.Name, inp.Value)
			addInParam(inparams, inp.Name, inp.Value)
		}
```

which force me to take the method out of the library as a workaround during migrating to csi-proxy to WMI based implementation.
Refer to https://github.com/kubernetes-csi/csi-proxy/pull/360/files#diff-90b6f12c56e11a97a650f55b092ccee8d80ff2e4cb2294394838160aaf39140aR95

This is the backtrace.

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x1752ffa]

goroutine 8 [running]:
testing.tRunner.func1.2({0x1888e60, 0x1f46e40})
	C:/Program Files/Go/src/testing/testing.go:1632 +0x489
testing.tRunner.func1()
	C:/Program Files/Go/src/testing/testing.go:1635 +0x669
panic({0x1888e60?, 0x1f46e40?})
	C:/Program Files/Go/src/runtime/panic.go:785 +0x136
github.com/go-ole/go-ole.(*IDispatch).VTable(0x0)
	C:/Users/Administrator/csi-proxy/vendor/github.com/go-ole/go-ole/idispatch.go:18 +0x1a
github.com/go-ole/go-ole.getIDsOfName(0x0, {0xc000141350, 0x1, 0x1})
	C:/Users/Administrator/csi-proxy/vendor/github.com/go-ole/go-ole/idispatch_windows.go:21 +0x232
github.com/go-ole/go-ole.(*IDispatch).GetIDsOfName(0x0, {0xc000141350, 0x1, 0x1})
	C:/Users/Administrator/csi-proxy/vendor/github.com/go-ole/go-ole/idispatch.go:22 +0x89
github.com/go-ole/go-ole.(*IDispatch).GetSingleIDOfName(0x0, {0x19982a3, 0xe})
	C:/Users/Administrator/csi-proxy/vendor/github.com/go-ole/go-ole/idispatch.go:47 +0xe7
github.com/go-ole/go-ole.(*IDispatch).InvokeWithOptionalArgs(0x0, {0x19982a3, 0xe}, 0x1, {0x0, 0x0, 0x0})
	C:/Users/Administrator/csi-proxy/vendor/github.com/go-ole/go-ole/idispatch.go:63 +0x85
github.com/go-ole/go-ole/oleutil.CallMethod(0x0, {0x19982a3, 0xe}, {0x0, 0x0, 0x0})
	C:/Users/Administrator/csi-proxy/vendor/github.com/go-ole/go-ole/oleutil/oleutil.go:51 +0x9f
github.com/microsoft/wmi/pkg/wmiinstance.(*WmiMethod).Execute(0xc00011e080, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0})
	C:/Users/Administrator/csi-proxy/vendor/github.com/microsoft/wmi/pkg/wmiinstance/WmiMethod.go:84 +0x6c5
github.com/kubernetes-csi/csi-proxy/pkg/cim.InvokeCimMethod2({0x19a8a48, 0x1e}, {0x199c653, 0x13}, {0x199f7eb, 0x16}, 0x0)
	C:/Users/Administrator/csi-proxy/pkg/cim/wmi.go:225 +0x8fd
```

The app panic at the the following line.

```go
	inparamsRaw, err := rawMethod.ToIDispatch().GetProperty("InParameters")
	if err != nil {
		return nil, err
	}
	defer inparamsRaw.Clear()

>>>	inparams, err := oleutil.CallMethod(inparamsRaw.ToIDispatch(), "SpawnInstance_")
	if err != nil {
		return nil, err
	}
	defer inparams.Clear()
```

For such WMI method which expects no `InParameters` GetProperty returns a VT_VARIANT NULL in this case. The code didn't check for the case and panic when trying to call the `IDispatch.SpawnInstance_`.

--- a/pkg/wmiinstance/WmiMethod.go
+++ b/pkg/wmiinstance/WmiMethod.go
@@ -75,27 +75,36 @@ func (c *WmiMethod) Execute(inParam, outParam WmiMethodParamCollection) (result 
 	}
 	defer rawMethod.Clear()
 
-	inparamsRaw, err := rawMethod.ToIDispatch().GetProperty("InParameters")
-	if err != nil {
-		return nil, err
-	}
-	defer inparamsRaw.Clear()
+	params := []interface{}{c.Name}
 
-	inparams, err := oleutil.CallMethod(inparamsRaw.ToIDispatch(), "SpawnInstance_")
-	if err != nil {
-		return nil, err
-	}
-	defer inparams.Clear()
+	if len(inParam) > 0 {
+		inparamsRaw, err := rawMethod.ToIDispatch().GetProperty("InParameters")
+		if err != nil {
+			return nil, err
+		}
+		defer inparamsRaw.Clear()
 
-	for _, inp := range inParam {
-		// 	log.Printf("InParam [%s]=>[%+v]\n", inp.Name, inp.Value)
-		c.addInParam(inparams, inp.Name, inp.Value)
+		// Method with no parameters may return a VARIANT with nil IDispatch
+		if inparamsRaw.Val != 0 {
+			inparams, err := oleutil.CallMethod(inparamsRaw.ToIDispatch(), "SpawnInstance_")
+			if err != nil {
+				return nil, err
+			}
+			defer inparams.Clear()
+
+			for _, inp := range inParam {
+				// 	log.Printf("InParam [%s]=>[%+v]\n", inp.Name, inp.Value)
+				c.addInParam(inparams, inp.Name, inp.Value)
+			}
+
+			params = append(params, inparams)
+		}
 	}
 
 	result = &WmiMethodResult{
 		OutMethodParams: map[string]*WmiMethodParam{},
 	}
-	outparams, err := c.classInstance.GetIDispatch().CallMethod("ExecMethod_", c.Name, inparams)
+	outparams, err := c.classInstance.GetIDispatch().CallMethod("ExecMethod_", params...)
 	if err != nil {
 		return
 	}

